plugins/org.jboss.tools.quarkus.lsp4e/src/org/jboss/tools/quarkus/lsp4e/QuarkusLanguageClient.java

```diff
@@ -65,20 +65,11 @@
  */
 public class QuarkusLanguageClient extends LanguageClientImpl implements MicroProfileLanguageClientAPI {
 
-	private static IMicroProfilePropertiesChangedListener SINGLETON_LISTENER;
-
 	private IMicroProfilePropertiesChangedListener listener = event -> {
 		((MicroProfileLanguageServerAPI) getLanguageServer()).propertiesChanged(event);
 	};
 
 	public QuarkusLanguageClient() {
-		// FIXME : how to remove the listener????
-		// The listener should be removed when language server is shutdown, how to
-		// manage that????
-		if (SINGLETON_LISTENER != null) {
-			MicroProfileCorePlugin.getDefault().removeMicroProfilePropertiesChangedListener(SINGLETON_LISTENER);

```

Comment:
Thats very good to remove this ugly code, but it is very important to remove the listener. Since last version of LSP4E, the LanguageClient defines a dispose method called when the language server is stopped, so I suggest that you add the method:

```java
@Override
public void dispose() {
   super.dispose();
  MicroProfileCorePlugin.getDefault().removeMicroProfilePropertiesChangedListener(listener);
}
```